tests/network/services.go

```diff
@@ -247,51 +237,35 @@ var _ = SIGDescribe("Services", func() {
 		})
 
 		Context("with a service matching the vmi exposed", func() {
-			var jobsToCleanup []func() error
-			var serviceManager *vmiServiceManager
-			var serviceNamePrefix string
-
-			BeforeEach(func() {
-				serviceNamePrefix = "myservice"
-				serviceManager = newVMIServiceManager(inboundVMI, servicePort, serviceNamePrefix, selectorLabelKey, selectorLabelValue)
-
-				services := serviceManager.buildK8sServicesSpec()
-				Expect(services).NotTo(BeEmpty(), "a service should be exposed per each iface. At least one *must* be present.")
-
-				for _, exposedService := range services {
-					_, err := virtClient.CoreV1().Services(inboundVMI.Namespace).Create(&exposedService)
-					Expect(err).NotTo(HaveOccurred())
+			table.DescribeTable("should be able to reach the vmi based on labels specified on the vmi", func(ipFamily k8sv1.IPFamily) {
+				var service *k8sv1.Service
+
+				serviceName := "myservice"
+				if ipFamily == k8sv1.IPv6Protocol {
+					serviceName = serviceName + "v6"
+					service = buildIPv6ServiceSpec(serviceName, servicePort, servicePort, selectorLabelKey, selectorLabelValue)
+				} else {
+					service = buildServiceSpec(serviceName, servicePort, servicePort, selectorLabelKey, selectorLabelValue)
 				}
 
-				// reset the job cleanup functions list
-				jobsToCleanup = nil
-			})
-
-			AfterEach(func() {
-				var errors []error
-				for _, exposedService := range serviceManager.services {
-					err := virtClient.CoreV1().Services(inboundVMI.Namespace).Delete(exposedService.Name, &k8smetav1.DeleteOptions{})
-					errors = append(errors, err)
-				}
-				for _, err := range errors {
-					Expect(err).NotTo(HaveOccurred())
-				}
-			})
-
-			AfterEach(func() {
-				cleanupJobs(jobsToCleanup)
-			})
+				_, err := virtClient.CoreV1().Services(inboundVMI.Namespace).Create(service)
+				Expect(err).NotTo(HaveOccurred(), "the k8sv1.Service entity should have been created.")
 
-			It("should be able to reach the vmi based on labels specified on the vmi", func() {
-				for _, exposedService := range serviceManager.services {
-					jobCleanupFunc := assertConnectivityToService(exposedService.Name, inboundVMI.Namespace, servicePort)
-					jobsToCleanup = append(jobsToCleanup, jobCleanupFunc)
-				}
-			})
+				jobCleanupFunc := assertConnectivityToService(serviceName, inboundVMI.Namespace, servicePort)
+				Expect(jobCleanupFunc()).To(Succeed(), "cleaning up the k8sv1.Service entity should have succeeded.")
+			},
+				table.Entry("when the service is exposed by an IPv4 address.", k8sv1.IPv4Protocol),
+				table.Entry("when the service is exposed by an IPv6 address.", k8sv1.IPv6Protocol),
+			)
 
 			It("should fail to reach the vmi if an invalid servicename is used", func() {
-				jobCleanupFunc := assertNoConnectivityToService("wrongservice", inboundVMI.Namespace, servicePort)
-				jobsToCleanup = append(jobsToCleanup, jobCleanupFunc)
+				serviceName := "wrongservice"
```

Comment:
Why are you creating a service? Isn't the purpose of this test is to test connectivity to non-existing service?

Reply:
/me facepalms.

Really good catch; thanks for pointing this out.